src/Adapter/MSTest.TestAdapter/ObjectModel/TestMethod.cs

```diff
@@ -28,20 +29,23 @@ public sealed class TestMethod : ITestMethod
     private string? _declaringClassFullName;
     private string? _declaringAssemblyName;
 
+    private static readonly ConcurrentDictionary<int, object?[]> DataDictionary = new();
```

Comment:
Notes:

- This is bad. The dictionary in theory can grow up indefinitely.
- This PR is mostly to open a discussion about the possibilities to improve the situation for some parameterized tests broken scenarios.

@Evangelink thoughts? I feel like some of the parameterized test issues could be fixable by trying to move forward in this direction without the need for breaking changes?